src/main/java/com/lambdaworks/redis/SimpleSerializers/JavaSerializer.java

```diff
@@ -0,0 +1,80 @@
+package com.lambdaworks.redis.SimpleSerializers;
+
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.io.*;
+
+/**
+ * @author <a href="mailto:a.abdelfatah@live.com">Ahmed Kamal</a>
+ * @since 12.01.16 09:17
+ */
+
+public class JavaSerializer {
+
+    protected static final InternalLogger logger = InternalLoggerFactory.getInstance(JavaSerializer.class);
+
+    public byte[] serializeObject(Object object) throws IOException {
+
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
```

Comment:
There's way too much try/catch in here. Check out try-with-resources (Java 1.7+) or Google Guava's [Closer](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/io/Closer.html). A cleaner approach is having a try/catch with an inner try/finally (closing the resources) block. The try/catch could then wrap any checked exception within a `RedisException`.